### DISCUSSION TITLE
Source vs. documentation discrepancy

### DISCUSSION THREAD

**Discussion by BalashovK** [microsoft/semantic-kernel#4342] [General]
Documentation:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.semantickernel.kernelbuilder?view=semantic-kernel-dotnet
Source:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Abstractions/KernelBuilder.cs

The difference is: Kernel Builder is internal (vs. public)

It looks like in older versions like 1.0.0-beta8 it was public.

Are you trying to discourage people from using Semantic Kernel from outside? Why?
Is there a financial reason for it (like, force users to pay for OpenAI LLM API by blocking free LlamaSharp integration) or was it just short-sighted software engineering decision?

Please either roll back this change or publish a workaround.

**Comment by stephentoub** [comment]
> Are you trying to discourage people from using Semantic Kernel from outside? Why?
Is there a financial reason for it (like, force users to pay for OpenAI LLM API by blocking free LlamaSharp integration) or was it just short-sighted software engineering decision?

Why all the hostility?

The docs need to be updated. Use Kernel.CreateBuilder, which returns an IKernelBuilder.

**Comment by evchaki** [comment]
@BalashovK - the documentation has been updated to 1.0.1 - https://learn.microsoft.com/en-us/dotnet/api/microsoft.semantickernel?view=semantic-kernel-dotnet.
